GitUI/GitUICommands.cs

```diff
@@ -478,41 +478,31 @@ bool Action()
             return DoActionOnRepo(owner, false, true, null, null, Action);
         }
 
-        public bool StartPullDialogAndPullImmediately(IWin32Window owner = null, string remoteBranch = null, string remote = null, bool fetchAll = false, bool prune = false)
+        public bool StartPullDialogAndPullImmediately(IWin32Window owner = null, string remoteBranch = null, string remote = null, AppSettings.PullAction pullAction = AppSettings.PullAction.None)
         {
-            return StartPullDialogAndPullImmediately(out _, owner, remoteBranch, remote, fetchAll, prune);
+            return StartPullDialogAndPullImmediately(out _, owner, remoteBranch, remote, pullAction);
         }
 
         /// <param name="pullCompleted">true if pull completed with no errors</param>
         /// <returns>if revision grid should be refreshed</returns>
-        public bool StartPullDialogAndPullImmediately(out bool pullCompleted, IWin32Window owner = null, string remoteBranch = null, string remote = null, bool fetchAll = false, bool prune = false)
+        public bool StartPullDialogAndPullImmediately(out bool pullCompleted, IWin32Window owner = null, string remoteBranch = null, string remote = null, AppSettings.PullAction pullAction = AppSettings.PullAction.None)
         {
-            return StartPullDialogInternal(owner, pullOnShow: true, out pullCompleted, remoteBranch, remote, fetchAll, prune);
+            return StartPullDialogInternal(owner, pullOnShow: true, out pullCompleted, remoteBranch, remote, pullAction);
         }
 
-        public bool StartPullDialog(IWin32Window owner = null, string remoteBranch = null, string remote = null, bool fetchAll = false, bool prune = false)
+        public bool StartPullDialog(IWin32Window owner = null, string remoteBranch = null, string remote = null, AppSettings.PullAction pullAction = AppSettings.PullAction.None)
         {
-            return StartPullDialogInternal(owner, pullOnShow: false, out _, remoteBranch, remote, fetchAll, prune);
+            return StartPullDialogInternal(owner, pullOnShow: false, out _, remoteBranch, remote, pullAction);
         }
 
-        private bool StartPullDialogInternal(IWin32Window owner, bool pullOnShow, out bool pullCompleted, string remoteBranch, string remote, bool fetchAll, bool prune)
+        private bool StartPullDialogInternal(IWin32Window owner, bool pullOnShow, out bool pullCompleted, string remoteBranch, string remote, AppSettings.PullAction pullAction)
         {
             var pulled = false;
 
             bool Action()
             {
-                using (var formPull = new FormPull(this, remoteBranch, remote))
+                using (var formPull = new FormPull(this, remoteBranch, remote, pullAction))
                 {
-                    if (fetchAll)
-                    {
-                        formPull.SetForFetchAll();
-                    }
-
-                    if (prune)
-                    {
-                        formPull.SetPrune();
```

Comment:
Neither `SetForFetchAll` nor `SetPrune` no longer used outside of `FormPull` and thus can be removed completely, instructions they contain can be inlined at respective call sites